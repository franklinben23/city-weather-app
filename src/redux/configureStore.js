import { combineReducers, createStore, applyMiddleware } from 'redux';
import Thunk from 'redux-thunk';
import logger from 'redux-logger';
import CityWeather from './cities/cities';

const rootReducer = combineReducers({
  cityWeathers: CityWeather,
});

const store = createStore(rootReducer, applyMiddleware(Thunk, logger));

export default store;
