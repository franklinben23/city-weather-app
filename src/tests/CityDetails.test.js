import TestRenderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import CityDetailsM from './__Mocks__/CityDetailsMock';
import '@testing-library/jest-dom';
import configureStore from '../redux/configureStore';

describe('Testing Mission', () => {
  it('Testing Mission component snapshot', () => {
    const cityObj = {
      cityCountry: "MX",
      cityImage: "https://www.elsoldemexico.com.mx/metropoli/cdmx/58gppj-cdmx-â€¦ntro%20de%20las%20100%20mejores%20ciudades%20del%20mundo.png",
      cityName: "Ciudad de Mexico",
      description: "broken clouds",
      humidity: 22,
      icon: "http://openweathermap.org/img/wn/04d@2x.png",
      id: 3530597,
      mainDesc: "Clouds",
      pressure: 1016,
      temp: 80.19,
      tempFeelsLike: 78.8,
      visibility: 10000
    }
    const testRenderer = TestRenderer.create(
      <Provider store={configureStore}>
        <CityDetailsM city={cityObj}/>
      </Provider>,
    ).toJSON();
    expect(testRenderer).toMatchSnapshot();
  });
});
