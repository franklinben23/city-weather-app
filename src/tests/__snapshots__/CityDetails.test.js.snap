// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Mission Testing Mission component snapshot 1`] = `
<div
  className="details-section"
>
  <div
    className="details-image"
  >
    <img
      alt="city"
      className="details-img"
    />
  </div>
  <div
    className="details-section-details"
  >
    <div
      className="detail-line"
    >
      <h5
        className="details-city-name"
      />
    </div>
    <div
      className="detail-line"
    >
      <span
        className="details-span span1"
      />
      <span
        className="details-span span2"
      >
        <img
          alt="city-icon"
          className="details-icon"
        />
      </span>
    </div>
    <div
      className="detail-line"
    >
      <p
        className="temp-tag tag1"
      >
         Temperature: 
      </p>
      <p
        className="temp-tag tag2"
      >
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 12h7v2h-7v7h-2V8a4 4 0 0 1 4-4h7v2h-7a2 2 0 0 0-2 2v4zm-7.5-2a3.5 3.5 0 1 1 0-7 3.5 3.5 0 0 1 0 7zm0-2a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3z"
            />
          </g>
        </svg>
      </p>
    </div>
    <div
      className="detail-line"
    >
      <p
        className="temp-tag tag1"
      >
         Temperature feeling:
      </p>
      <p
        className="temp-tag tag2"
      >
         
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 12h7v2h-7v7h-2V8a4 4 0 0 1 4-4h7v2h-7a2 2 0 0 0-2 2v4zm-7.5-2a3.5 3.5 0 1 1 0-7 3.5 3.5 0 0 1 0 7zm0-2a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3z"
            />
          </g>
        </svg>
      </p>
    </div>
    <div
      className="detail-line"
    >
      <p
        className="temp-tag tag1"
      >
         Pressure:
      </p>
      <p
        className="temp-tag tag2"
      />
    </div>
    <div
      className="detail-line"
    >
      <p
        className="temp-tag tag1"
      >
         Air Humidity:
      </p>
      <p
        className="temp-tag tag2"
      >
         
      </p>
    </div>
    <div
      className="detail-line"
    >
      <p
        className="temp-tag tag1"
      >
         Visibility: 
      </p>
      <p
        className="temp-tag tag2"
      >
         
        Km
      </p>
    </div>
  </div>
</div>
`;
